### DISCUSSION TITLE
IOS-XE Flows, no DST IP

### DISCUSSION THREAD

**Discussion by kmorin18** [akvorado/akvorado#1068] [Q&A] [answered]
Hey!
We have some IOS-XE routers and configured as per the docs.
We are getting the right amount of Mb/s but there is no DST IP data being processed in Akvorado.
We only have the flows setup on the WAN / transit side.

Are we missing something? Add netflow data from customer side interfaces?

Thanks!
![flows](https://github.com/akvorado/akvorado/assets/64321661/6394eadb-808e-4a90-b39a-c9edb7c70de1)

**Comment by kmorin18** [comment]
It seems like this would require BMP configuration
However BMP on IOS XE requires a neighbor address, as if they are expecting the BGP session to establish with Akvorado

**Reply by vincentbernat** [reply]
I don't think BMP would help. It seems more a configuration issue. Did you compare your configuration with the one in the documentation?

**Reply by kmorin18** [reply]
Thanks Vincent 

It seems like certain versions of IOS XE expect "export-protocol netflow-v9" instead of "version 9" under the flow exporter

Config changed, will report back for the public record/help

**Comment by kmorin18** [comment]
Weirdly still having this issue, despite having the proper flow record config:

```
flow record Akvorado
 match ipv4 tos
 match ipv4 protocol
 match ipv4 source address
 match ipv4 destination address
 match transport source-port
 match transport destination-port
 collect routing source as 4-octet
 collect routing destination as 4-octet
 collect routing next-hop address ipv4
 collect transport tcp flags
 collect interface output
 collect interface input
 collect counter bytes
 collect counter packets
 collect timestamp sys-uptime first
 collect timestamp sys-uptime last

```

**Reply by vincentbernat** [reply]
You could capture the received flows and check with Wireshark if the information is present or not.

**Reply by kmorin18** [reply]
It appears that the information is sent, but not the Netflow template, making decoding impossible at least in wireshark
We will keep digging as to why that is

**Comment by kmorin18** [comment]
I believe this was solved by adding a few items in the "collect" config:

flow record akvorado
 match ipv4 tos
 match ipv4 protocol
 match ipv4 source address
 match ipv4 destination address
 match transport source-port
 match transport destination-port
 match interface output
 match interface input
 collect routing next-hop address ipv4
 collect transport tcp flags
 collect counter bytes
 collect counter packets
 collect timestamp sys-uptime first
 collect timestamp sys-uptime last
 collect routing source as
 collect routing destination as

**Reply by kmorin18** [reply]
Spoke to fast, this didnt fix the issue
![image](https://github.com/user-attachments/assets/0217e3c6-7fa1-4e97-831d-0988563be528)

**Reply by kmorin18** [reply, marked as answer]
Seems like it works for SRC/DST IPs, not SRC/DST prefixes
![image](https://github.com/user-attachments/assets/c2d7a264-939d-4e28-852c-b59cfd5cd7d9)
